src/battle/action_cmd/water_block.c

```diff
@@ -28,14 +248,372 @@ ApiStatus func_802A9398_42A888(Evt* script, s32 isInitialCall) {
+            hud_element_set_render_pos(actionCommandStatus->hudElements[4], actionCommandStatus->hudElementX + 61, actionCommandStatus->hudElementY + 24);
+
+            id = actionCommandStatus->hudElements[10];
+            switch (actionCommandStatus->unk_64) {
+                case 0:
+                    hud_element_set_render_pos(id, actionCommandStatus->hudElementX + 91, actionCommandStatus->hudElementY + 7);
+                    break;
+                case 1:
+                    hud_element_set_render_pos(id, actionCommandStatus->hudElementX + 91, actionCommandStatus->hudElementY + 13);
+                    break;
+                case 2:
+                    hud_element_set_render_pos(id, actionCommandStatus->hudElementX + 91, actionCommandStatus->hudElementY + 13);
+                    break;
+            }
+
+            hud_element_set_render_pos(actionCommandStatus->hudElements[11], actionCommandStatus->hudElementX - 5, actionCommandStatus->hudElementY + 29);
+            hud_element_set_render_pos(actionCommandStatus->hudElements[12], actionCommandStatus->hudElementX + 7, actionCommandStatus->hudElementY + 29);
+            hud_element_set_render_pos(actionCommandStatus->hudElements[13], actionCommandStatus->hudElementX + 92, actionCommandStatus->hudElementY + 23);
+            return;
+        case 10:
+            btl_set_popup_duration(99);
+            if (actionCommandStatus->unk_4E != 0) {
+                actionCommandStatus->unk_4E--;
+                return;
+            }
+
+            actionCommandStatus->unk_54 = actionCommandStatus->unk_52 - 60;
+            if (actionCommandStatus->unk_54 < 42) {
+                actionCommandStatus->unk_54 = 42;
+            }
+            actionCommandStatus->state = 11;
+        case 11:
+            btl_set_popup_duration(99);
+            if (actionCommandStatus->unk_54 == 42) {
+                hud_element_set_script(actionCommandStatus->hudElements[11], &HudScript_TimingCharge3);
+                sfx_play_sound(SOUND_233);
+            }
+            if (actionCommandStatus->unk_54 == 22) {
+                hud_element_set_script(actionCommandStatus->hudElements[12], &HudScript_TimingCharge2);
+                sfx_play_sound(SOUND_233);
+            }
+            if (actionCommandStatus->unk_54 == 2) {
+                hud_element_set_script(actionCommandStatus->hudElements[2], &HudScript_TimingReady);
+                hud_element_set_script(actionCommandStatus->hudElements[0], &HudScript_AButtonDown);
+                sfx_play_sound(SOUND_234);
+            }
+            actionCommandStatus->unk_54--;
+            if (actionCommandStatus->unk_54 == 0) {
+                actionCommandStatus->unk_54 = 20;
+                actionCommandStatus->unk_5D = 0;
+                actionCommandStatus->state = 12;
+                actionCommandStatus->unk_5C = 0;
+                actionCommandStatus->unk_60 = 0;
+                return;
```

Comment:
Surprised that break != return in this func

Reply:
I fixed a few inconsistencies here